### ISSUE TITLE
Importing stl and transforming the mesh is slow

### ISSUE THREAD

**Issue by ortv2** [milmillin/passive-gripper#7]
I am facing the issue that importing an stl file and transforming the imported mesh is really slow. (as well as other operations, which require lots of computations).
For example, when I try to import the example stl file A4.stl, which has only 5KB, it takes around 40min until the mesh is fully imported.
Furthermore, CPU is constantly at 100% and the application window says that there is no response.

I am using:
Windows
Intel i7-11800H
32 GB RAM

I also tried it with a much better PC (Windows, 32 core CPU and 128 GB RAM) and it still takes ~40mins to import the file.

My questions: 
Have you faced similar issues or is it normal that this takes so long?
For which computers is the application designed? 

![Passive-gripper-noResponse](https://github.com/milmillin/passive-gripper/assets/156887601/5678f98f-1890-4093-9c98-05fae0d08fc3)

**Comment by LouYu2015** [comment]
On my laptop, it only takes around 10 seconds to load A4.stl. I'm using Intel i7-7700HQ on Ubuntu 22.04. I built the program at commit 79fe352.

I'm not familiar with the Windows environment. Here are some ideas you can try:

- Check build options to make sure that the program was compiled with O3 optimization instead of debugging mode
- Add the `-DENABLE_EASY_PROFILER=OFF` option to cmake

**Comment by LouYu2015** [comment]
I also tried to compile at commit 528141e and had no problem.

**Comment by ortv2** [comment]
Thanks for your reply!

Compiling it with the O3 optimization helped to reduce the import time to onyl a few seconds. When e.g. transforming the object or generate candidates the CPU usage is still at 100% and often the screen is greyed out saaying "no response"). 
This seems to be a Windows problem.

However, I managed to do all the steps until the Topy Config-File is generated. Topy successfully gives me the .vtk and .bin files but when I try to load one of the .bin files into your application is crashed (the window closes). I am getting the std::bad_alloc error.

**Comment by milmillin** [comment]
Yes, all operations are CPU intensive. It will use all cores possible, which might also block the ui thread. This is fine you just have to keep it running.

Could you try opening the .vtk file in ParaView?

**Comment by ortv2** [comment]
Yes, I could open the .vtk files in ParaView without any problems.

**Comment by milmillin** [comment]
Could you send me the .bin file?

**Comment by ortv2** [comment]
Thank you for your reply. 
I appended the .vkt and the .bin file in the zip directory.
[topy_results.zip]()

**Comment by ortv2** [comment]
As uploading a zip file containg the .bin file here in the github issue failed, I uploaded the .bin file to a repo. 
[Here you can find the it](https://github.com/ortv2/PassiveGripperFiles)

**Comment by ortv2** [comment]
Here I attached the .bin file and the topy configuration file.

[topy_results.zip](https://github.com/user-attachments/files/16709418/topy_results.zip)
